api/api.go

```diff
@@ -35,120 +30,115 @@ func HttpPOST(url string, data string) (resp *http.Response, err error) {
 	if srv.renter != nil {
-		srv.handleHTTPRequest(mux, "/hostdb/hosts/active", srv.renterHostsActiveHandler)
-		srv.handleHTTPRequest(mux, "/hostdb/hosts/all", srv.renterHostsAllHandler)
+		mux.HandlerFunc("GET", "/hostdb/hosts/active", srv.renterHostsActiveHandler)
+		mux.HandlerFunc("GET", "/hostdb/hosts/all", srv.renterHostsAllHandler)
 	}
 
 	// Miner API Calls
 	if srv.miner != nil {
-		srv.handleHTTPRequest(mux, "/miner", srv.minerHandler)                            // GET
-		srv.handleHTTPRequest(mux, "/miner/header", srv.minerHeaderHandler)               // GET, POST
-		srv.handleHTTPRequest(mux, "/miner/start", srv.minerStartHandler)                 // POST
-		srv.handleHTTPRequest(mux, "/miner/stop", srv.minerStopHandler)                   // POST
-		srv.handleHTTPRequest(mux, "/miner/headerforwork", srv.minerHeaderforworkHandler) // COMPATv0.4.8
-		srv.handleHTTPRequest(mux, "/miner/submitheader", srv.minerSubmitheaderHandler)   // COMPATv0.4.8
+		mux.GET("/miner", srv.minerHandler)
+		mux.GET("/miner/header", srv.minerHeaderHandlerGET)
+		mux.POST("/miner/header", srv.minerHeaderHandlerPOST)
+		mux.POST("/miner/start", srv.minerStartHandler)
+		mux.POST("/miner/stop", srv.minerStopHandler)
+		mux.GET("/miner/headerforwork", srv.minerHeaderHandlerGET) // COMPATv0.4.8
+		mux.GET("/miner/submitheader", srv.minerHeaderHandlerPOST) // COMPATv0.4.8
 	}
 
 	// Renter API Calls - Unfinished
 	if srv.renter != nil {
-		srv.handleHTTPRequest(mux, "/renter/downloadqueue", srv.renterDownloadqueueHandler)
-		srv.handleHTTPRequest(mux, "/renter/files/delete", srv.renterFilesDeleteHandler)
-		srv.handleHTTPRequest(mux, "/renter/files/download", srv.renterFilesDownloadHandler)
-		srv.handleHTTPRequest(mux, "/renter/files/list", srv.renterFilesListHandler)
-		srv.handleHTTPRequest(mux, "/renter/files/load", srv.renterFilesLoadHandler)
-		srv.handleHTTPRequest(mux, "/renter/files/loadascii", srv.renterFilesLoadAsciiHandler)
-		srv.handleHTTPRequest(mux, "/renter/files/rename", srv.renterFilesRenameHandler)
-		srv.handleHTTPRequest(mux, "/renter/files/share", srv.renterFilesShareHandler)
-		srv.handleHTTPRequest(mux, "/renter/files/shareascii", srv.renterFilesShareAsciiHandler)
-		srv.handleHTTPRequest(mux, "/renter/files/upload", srv.renterFilesUploadHandler)
-		srv.handleHTTPRequest(mux, "/renter/status", srv.renterStatusHandler)
+		mux.GET("/renter/downloadqueue", srv.renterDownloadqueueHandler)
+		mux.GET("/renter/files", srv.renterFilesHandler)
+		mux.POST("/renter/load", srv.renterLoadHandler)
+		mux.POST("/renter/loadascii", srv.renterLoadAsciiHandler)
+		mux.POST("/renter/rename", srv.renterRenameHandler)
+		mux.POST("/renter/delete/*path", srv.renterDeleteHandler)
+		mux.GET("/renter/download/*path", srv.renterDownloadHandler)
+		mux.GET("/renter/share/*path", srv.renterShareHandler)
+		mux.GET("/renter/shareascii/*path", srv.renterShareAsciiHandler)
+		mux.POST("/renter/upload/*path", srv.renterUploadHandler)
 	}
```

Comment:
btw, here are the docs for the dropbox API as a reference: https://www.dropbox.com/developers-v1/core/docs#files-GET